tests/Unit/PointwiseFunctions/AnalyticData/GrMhd/Test_MagnetizedTovStar.cpp

```diff
@@ -0,0 +1,80 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include "Domain/LogicalCoordinates.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "NumericalAlgorithms/LinearOperators/Divergence.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "PointwiseFunctions/AnalyticData/GrMhd/MagnetizedTovStar.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/RelativisticEuler/TovStar.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+
+SPECTRE_TEST_CASE("Unit.PointwiseFunctions.AnalyticData.GrMhd.MagTovStar",
+                  "[Unit][PointwiseFunctions]") {
+  const auto mag_tov_opts =
+      TestHelpers::test_creation<grmhd::AnalyticData::MagnetizedTovStar>(
+          "CentralDensity: 1.28e-3\n"
+          "PolytropicConstant: 100.0\n"
+          "PolytropicExponent: 2.0\n"
+          "PressureExponent: 2\n"
+          "VectorPotentialAmplitude: 2500\n"
+          "CutoffPressureFraction: 0.04\n");
+  const RelativisticEuler::Solutions::TovStar<gr::Solutions::TovSolution> tov{
+      1.28e-3, 100.0, 2.0};
+  const auto mag_tov = serialize_and_deserialize(mag_tov_opts);
```

Comment:
Should there be a check that the mag_tov from option parsing matches one from an explicit constructor call? this would check the option parsing and the operator==.